WooCommerce/Classes/POS/Presentation/CardReaderConnection/UI States/PointOfSaleLoadingView.swift

```diff
@@ -13,6 +15,25 @@ struct PointOfSaleLoadingView: View {
             .multilineTextAlignment(.center)
             Spacer()
         }
+        .onAppear {
+            trackTimeOnAppear()
+        }
+        .onDisappear {
+            trackElapsedTimeOnDisappear()
+        }
+    }
+}
+
+private extension PointOfSaleLoadingView {
+    func trackTimeOnAppear() {
+        appearTimeInMilliseconds = Date().timeIntervalSince1970 * Constants.toMilliseconds
+    }
+
+    func trackElapsedTimeOnDisappear() {
+        if let appearTimeInMilliseconds = appearTimeInMilliseconds {
+            let elapsedTimeInMilliseconds = (Date().timeIntervalSince1970 * Constants.toMilliseconds) - appearTimeInMilliseconds
```

Comment:
Used `.now()` and `.uptimeNanoseconds` then divide by 1_000_000 initially, but AI says that TimeInterval + multiplying it by 1000 is closer to Android's implementation. I have no special preference 🤔.